src/main/java/frc/robot/subsystems/amp/AmpIOSim.java

```diff
@@ -0,0 +1,140 @@
+package frc.robot.subsystems.amp;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.simulation.FlywheelSim;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import frc.robot.Constants;
+
+public class AmpIOSim implements AmpIO {
+  private final SingleJointedArmSim sim =
+      new SingleJointedArmSim(
+          DCMotor.getKrakenX60(1),
+          67.5, // gearing
+          0.192383865, // MOI
+          0.3, // arm length
+          Units.degreesToRadians(0), // min angle -- floor
+          Units.degreesToRadians(180), // max angle -- hard stop
+          false,
+          Units.degreesToRadians(0));
+  private final PIDController pid = new PIDController(0.0, 0.0, 0.0);
+
+  private FlywheelSim spinnerSim = new FlywheelSim(DCMotor.getFalcon500(2), 1.5, 0.004);
+  private PIDController spinnerPid = new PIDController(0.0, 0.0, 0.0);
+  private SimpleMotorFeedforward ff;
+
+  private boolean closedLoop = false;
+  private double appliedVolts = 0.0;
+  public double setPoint = 0.0;
+
+  private boolean closedSpinnerLoop = false;
+  private double ffVolts = 0.0;
+  private double appliedSpinnerVolts = 0.0;
+  public double speedPoint = 0.0;
+
+  @Override
+  public void updateInputs(AmpIOInputs inputs) {
+    if (closedLoop) {
+      // Need this bc of L sysID that uses open loop
```

Comment:
better method of just setting the applied volts in each set speed/voltage function. Looks better/makes code more readable. U can do if u want (I would). See climber in main to fix if u want